index.html

```diff
@@ -4116,91 +4060,475 @@ <h3>Parsing content</h3>
+        <dd>
+          This is a generic problem with all existing NFC tags.
+          The data is considered application specific.
+          Implementations need security hardening.
+          Involuntary touch is low probability due to short range and critical
+          angle for reading, and due to the focus requirements.
+          Automatic actions for smart posters and other tags should not be
+          allowed. The user must be made aware and given the ability to control
+          what is happening during the NFC communication. For instance, opening
+          content from <a>smart poster</a>, automatic connection to (possibly
+          malicious) WiFi via <a>NFC handover</a>, etc.
+          Do no allow actions from untrusted NFC tags, trust can be established
+          via the <a>NDEF signature</a> check.
+        </dd>
+      </dl>
+    </section>
+  </section>
+
+  <section class="informative"> <h2>Security mechanisms</h2>
+    <p>
+      In order to support security policies concerning Web NFC, the following
+      mechanisms are recommended for the implementations.
+    </p>
+    <section> <h3><dfn>Obtaining permission</dfn></h3>
+      Implementations SHOULD use one or more of the following mechanisms to
+      <dfn>obtain permission</dfn>.
+      <ul>
+        <li>
+          A mechanism for obtaining <a>expressed permission</a> of the user.
+        </li>
+        <li>
+          A mechanism for <a>ask for forgiveness</a> policies.
+        </li>
+      </ul>
+      <section> <h4><dfn>Expressed permission</dfn></h4>
+        <p>
+          Denotes an explicit permission given by the user.
+          The [[[PERMISSIONS]]] API is suggested to be used by
+          UAs for implementing NFC related permissions in order to minimize
+          the need for user prompting.
+        </p>
+        <p>
+          The
+          <a href="https://www.w3.org/TR/permissions/#dictdef-permissiondescriptor">
+          <dfn>Web NFC permission name</dfn></a> is
+          <a href="https://github.com/w3c/permissions/issues/47">defined</a> as
+          "`nfc`".
+        </p>
+      </section>
+      <section> <h4><dfn>Ask for forgiveness</dfn></h4>
+        <p>
+          Denotes an unobtrusive way to notify the user about an ongoing operation
+          with a possible action to cancel or undo the operation, and to
+          handle future occurences of the same operation by the same action.
+        </p>
+      </section>
+    </section>
+
+    <section> <h3>Encrypting <a>NFC content</a></h3>
+      <p>
+        For trusting the confidentiality of the data exchanged via NFC, user agents
+        MAY use encrypted <a>NFC content</a> with key management based on
+        Public Key Infrastructure (PKI).
+        Key management is out of the scope of Web NFC.
+      </p>
+      <p class="note">
+        This is different from application level end to end encryption.
+        Key management is then in application domain.
```

Comment:
Can we be explicit and define what/who is application domain?

Reply:
Actually the note does not need the sentence "Key management is then in application domain."

Reply:
This case applies to the implementation. I think it's not realistic to Web NFC to promise implementing content encryption with key management, so we could as well remove it.

However, the security section is a general informal section, and the question is should we _exclude_ the possibility any implementation goes with this?

Reply:
I would not exclude the possibility but rather unclutter the security section?
Maybe add that to the security questionnaire for TAG.